File: svc-data/src/main/java/com/beancounter/marketdata/batch/ScheduledValuation.java
package com.beancounter.marketdata.batch;

import com.beancounter.marketdata.providers.PriceRefresh;
import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.EnableTransactionManagement;

@EnableScheduling
@EnableTransactionManagement

@Service
@Slf4j
public class ScheduledValuation {
  private final PriceRefresh priceRefresh;

  public ScheduledValuation(PriceRefresh priceRefresh) {
    this.priceRefresh = priceRefresh;
  }

  @Scheduled(cron = "${beancounter.assets.schedule:0 */30 7-18 ? * Tue-Sat}")
  public void updatePrices() {
    priceRefresh.updatePrices();
  }

}

File: svc-data/src/main/java/com/beancounter/marketdata/providers/ScheduledValuation.kt
package com.beancounter.marketdata.providers

import com.beancounter.common.utils.DateUtils
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.EnableTransactionManagement

@EnableTransactionManagement
@Service
class ScheduledValuation(private val priceRefresh: PriceRefresh) {
    lateinit var dateUtils: DateUtils


    companion object {
        private val log = LoggerFactory.getLogger(ScheduledValuation::class.java)
    }

}